### PR TITLE
aks-engine: Add version 0.50.0

### LINKED ISSUE #1014: [Request] Add aks-engine into repo
AKS Engine (https://github.com/Azure/aks-engine) is the easiest way to provision a self-managed Kubernetes cluster on Azure. It provides convenient tooling to quickly bootstrap Kubernetes clusters on Azure. By leveraging ARM (Azure Resource Manager), AKS Engine helps you create, destroy and maintain clusters provisioned with basic IaaS resources in Azure. AKS Engine is also the library used by AKS for performing these operations to provide managed service implementations.

And AKS engine is a single exe without dependency. Which can add it easily. I raise a PR for it.

### PR DESCRIPTION
**Pull request by elemount**
Resolve #1014

### CODE REVIEW

File: bucket/aks-engine.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            "url": "https://github.com/Azure/aks-engine/releases/download/v0.49.0/aks-engine-v0.49.0-windows-amd64.zip",
+            "hash": "e288f8f36972d74a920ac749559bd28691a0d1570e1de3ebaa8b3f5a7260ae33"
+        }
+    },
+    "bin": "aks-engine.exe",
+    "checkver": "github",
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/Azure/aks-engine/releases/download/v$version/aks-engine-v$version-windows-amd64.zip"
```
Comment (Ash258) [inline bucket/aks-engine.json:17]:
```suggestion
                "url": "https://github.com/Azure/aks-engine/releases/download/v$version/aks-engine-v$version-windows-amd64.zip",
                "extract_dir": "aks-engine-v$version-windows-amd64"
```

File: bucket/aks-engine.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "description": "AKS Engine is the easiest way to provision a self-managed Kubernetes cluster on Azure.",
+    "homepage": "https://github.com/Azure/aks-engine",
+    "version": "0.49.0",
+    "license": "MIT",
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/Azure/aks-engine/releases/download/v0.49.0/aks-engine-v0.49.0-windows-amd64.zip",
+            "hash": "e288f8f36972d74a920ac749559bd28691a0d1570e1de3ebaa8b3f5a7260ae33"
```
Comment (Ash258) [inline bucket/aks-engine.json:9]:
```suggestion
            "url": "https://github.com/Azure/aks-engine/releases/download/v0.50.0/aks-engine-v0.50.0-windows-amd64.zip",
            "hash": "bd4d4096dbccac769838f50977206e3b5ef4ca79c2e1946fdf568090418e6de0",
            "extract_dir": "aks-engine-v0.50.0-windows-amd64"
```

File: bucket/aks-engine.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+    "description": "AKS Engine is the easiest way to provision a self-managed Kubernetes cluster on Azure.",
+    "homepage": "https://github.com/Azure/aks-engine",
+    "version": "0.49.0",
+    "license": "MIT",
```
Comment (Ash258) [inline bucket/aks-engine.json:5]:
```suggestion
    "version": "0.50.0",
    "description": "Self-managed Kubernetes cluster provisioning on Azure",
    "homepage": "https://github.com/Azure/aks-engine",
    "license": "MIT",
```

**Comment by elemount** [comment]
Hi @Ash258 , I've applied the suggestions from the code review :)

**Pull request merged**